pex/third_party/__init__.py

```diff
@@ -0,0 +1,392 @@
+      prefix = os.path.join(prefix, os.path.basename(path))
+      path = os.path.dirname(path)
+    raise ValueError('Could not find the zip file housing {}'.format(root))
+
+  def iter_root_modules(self, relpath):
+    for package in self._filter_names(relpath, r'(?P<module>[^/]+)\.py', 'module'):
+      if package != '__init__':
+        yield package
+
+  def iter_root_packages(self, relpath):
+    for package in self._filter_names(relpath, r'(?P<package>[^/]+)/__init__\.py', 'package'):
+      yield package
+
+  def _filter_names(self, relpath, pattern, group):
+    pat = re.compile(r'^{prefix}{pattern}$'
+                     .format(prefix=self.prefix + ((relpath + os.sep) if relpath else ''),
+                             pattern=pattern))
+
+    with contextlib.closing(zipfile.ZipFile(self.zipfile_path)) as zf:
+      for name in zf.namelist():
+        match = pat.match(name)
+        if match:
+          yield match.group(group)
+
+
+class VendorImporter(object):
+  """A `PEP-302 <https://www.python.org/dev/peps/pep-0302/>`_ meta_path importer for vendored code.
+
+  This importer redirects imports from its package to vendored code, optionally exposing the
+  vendored code by its un-prefixed module name as well.
+
+  For example, if the ``requests`` distribution was vendored, it could be imported using this
+  importer via ``import pex.third_party.requests`` as longs as:
+
+    * The requests distribution was housed under some importable path prefix inside the this
+      distribution.
+    * The requests distribution had its self-referential absolute imports re-written to use the
+      vendored import prefix.
+  """
+
+  @staticmethod
+  def _abs_root(root=None):
+    from pex import vendor
+
+    return os.path.abspath(root or vendor.VendorSpec.ROOT)
+
+  @classmethod
+  def _iter_importables(cls, root, path_items, prefix):
+    module_iterator = _DirIterator(root) if os.path.isdir(root) else _ZipIterator.containing(root)
+    for path_item in path_items:
+      for module_name in module_iterator.iter_root_modules(path_item):
+        yield _Importable(module=module_name, is_pkg=False, path=path_item, prefix=prefix)
+      for package_name in module_iterator.iter_root_packages(path_item):
+        yield _Importable(module=package_name, is_pkg=True, path=path_item, prefix=prefix)
+
+  @classmethod
+  def _iter_all_installed_vendor_importers(cls):
+    for importer in sys.meta_path:
+      if isinstance(importer, cls):
+        yield importer
+
+  @classmethod
+  def _iter_installed_vendor_importers(cls, prefix, root, path_items):
+    for importer in cls._iter_all_installed_vendor_importers():
+      # All Importables for a given VendorImporter will have the same prefix.
+      if importer._importables and importer._importables[0].prefix == prefix:
+        if importer._root == root:
+          if {importable.path for importable in importer._importables} == set(path_items):
```

Comment:
Are you nesting the conditionals for readability? Because there are no else clauses, I think one multiline if with multiple `and`s could express intent better.

Reply:
Expressing intent and readability seem to be synonyms to me! Yes - this is for readability. The single expression form is where I started and it seemed both not easily parsed through and left no natural place for the comment on the 1st condition noting any `_Importable`'s prefix would do.